### PR TITLE
[ISSUE-379] Integrate e2e tests with CSI Operator

### PR DESCRIPTION
**Pull request by safronovD**
### Issue #379

Update CSI deployment process in e2e testing stage:
- Install CSI Operator and Deployment from helm charts
- Delete `csibmnodes` between stages
- Change logic to create custom drives -> 1) Set defaultDrivesCount=0 2) Update configmap
- Refactor `nodeLabels` test
- fix test suites

## PR checklist
- [x] Add link to the issue
- [x] Choose PR label
- [ ] New unit tests added
- [x] Modified code has meaningful comments
- [ ] All TODOs are linked with the issues
- [ ] All comments are resolved

## Testing
_Link to custom CI build_

### CODE REVIEW

File: test/e2e/baremetal_e2e_test.go [hunk 1]
```diff
@@ -18,6 +18,7 @@ package main
 
 import (
 	"flag"
+
```
Comment (yurkov-anton) [inline test/e2e/baremetal_e2e_test.go:21]: is it need?

**Review by yurkov-anton — APPROVED** [review]

File: test/e2e/common/scheduler-restart.go [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+	return &schedulerRestartChecker{
+		IsInitialized:      false,
+		c:                  client,
+		schedulerLabel:     "component=kube-scheduler",
+		restartWaitTimeout: time.Minute * 2,
+	}
+}
+
+type schedulerRestartChecker struct {
+	c                  clientset.Interface
```
Comment (AnastasiaShemyakinskaya) [inline test/e2e/common/scheduler-restart.go:39]: Not really intuitive name

File: test/e2e/common/scheduler-restart.go [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+func (rc *schedulerRestartChecker) getPODStartTimeMap() (map[string]metav1.Time, error) {
+	pods, err := rc.findSchedulerPods()
+	if err != nil {
+		return nil, err
+	}
+	return rc.buildPODStartTimeMap(pods), nil
+}
+
+func (rc *schedulerRestartChecker) buildPODStartTimeMap(pods *corev1.PodList) map[string]metav1.Time {
+	data := map[string]metav1.Time{}
```
Comment (AnastasiaShemyakinskaya) [inline test/e2e/common/scheduler-restart.go:120]: Imo it's better `data = make( map[string]metav1.Time{}, len(pods.Items))`

File: test/e2e/scenarios/different-sc-test.go [hunk 4]
```diff
@@ -60,21 +60,28 @@ func differentSCTypesTest(driver testsuites.TestDriver) {
...
+			driverType = driveTypeSSD
+		} else {
+			driverType = driveTypeHDD
 		}
-		perTestConf, driverCleanup = driver.PrepareTest(f)
 
-		k8sSC = driver.(*baremetalDriver).GetStorageClassWithStorageType(perTestConf, scType)
+		nodes, err := e2enode.GetReadySchedulableNodesOrDie(f.ClientSet)
+		framework.ExpectNoError(err)
+		var nodeNames []string
```
Comment (AnastasiaShemyakinskaya) [inline test/e2e/scenarios/different-sc-test.go:76]: The same comment^ ` nodeNames = make([]string, len(nodes.Items))`

File: test/e2e/scenarios/node-labels.go [hunk 5]
```diff
@@ -132,7 +87,36 @@ func labeledDeployTestSuite(driver testsuites.TestDriver) {
...
+
+	return workerNodes
+}
+
+func cleanNodeLabels(c clientset.Interface) {
+	nodes := getWorkerNodes(c)
+	for _, node := range nodes {
+		if _, ok := node.Labels[label]; ok {
+			delete(node.Labels, label)
+
```
Comment (AnastasiaShemyakinskaya) [inline test/e2e/scenarios/node-labels.go:116]: Empty line is not needed

File: Makefile.validation [hunk 6]
```diff
@@ -49,6 +49,17 @@ test-sanity:
 #	should fail when requesting to create a volume with already existing name and different capacity|\
 #	should not fail when requesting to create a volume with already existing name and same capacity" -ginkgo.v -timeout=0
 
+load-helm-charts:
```
Comment (mishoyama) [inline Makefile.validation:52]: This target must be in private repository.

File: Makefile.validation [hunk 7]
```diff
@@ -36,8 +36,8 @@ pr-validation-junit:
 
 # Run e2e tests for CI. All of these tests use ginkgo so we can use native ginkgo methods to print junit output.
 # Also go test doesn't provide functionnality to save test's log into the file. Use > to archieve artifatcs.
-test-ci:
-	${GO_ENV_VARS} CI=true go test -v test/e2e/baremetal_e2e_test.go -ginkgo.v -ginkgo.progress -kubeconfig=${HOME}/.kube/config -timeout=0 > log.txt
+test-ci-with-operator:
```
Comment (mishoyama) [inline Makefile.validation:39]: Why did you rename?

**Review by mishoyama — CHANGES_REQUESTED** [review]

File: test/e2e/baremetal_e2e_test.go (hunk 1 above)
Comment (safronovD) [reply test/e2e/baremetal_e2e_test.go]: deleted

File: test/e2e/common/scheduler-restart.go (hunk 2 above)
Comment (safronovD) [reply test/e2e/common/scheduler-restart.go]: renamed `client`

File: test/e2e/common/scheduler-restart.go (hunk 3 above)
Comment (safronovD) [reply test/e2e/common/scheduler-restart.go]: fixed

File: test/e2e/scenarios/different-sc-test.go (hunk 4 above)
Comment (safronovD) [reply test/e2e/scenarios/different-sc-test.go]: fixed

File: test/e2e/scenarios/node-labels.go (hunk 5 above)
Comment (safronovD) [reply test/e2e/scenarios/node-labels.go]: deleted

File: Makefile.validation (hunk 6 above)
Comment (safronovD) [reply Makefile.validation]: removed

File: Makefile.validation (hunk 7 above)
Comment (safronovD) [reply Makefile.validation]: It was needed for testing in CI. Renamed back

File: test/e2e/common/deploy.go [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"fmt"
+
```
[the remark this answers is not in the document]
Comment (safronovD) [reply test/e2e/common/deploy.go]: deleted

**Review by mishoyama — APPROVED** [review]

**Review by AnastasiaShemyakinskaya — APPROVED** [review]

**Review by painhardcore — APPROVED** [review]
LGTM
